src/tasks/shop-admin/Product/BulkEditProducts.ts

```diff
@@ -0,0 +1,123 @@
+                    await AdminProductBulkEdit.changeRestockTimeCheckbox.click();
+                    await AdminProductBulkEdit.restockTimeChangeMethodDropdown.click();
+                    await (await AdminProductBulkEdit.getDropdownEntry(changes['restockTime'].method)).click();
+                    if (changes['restockTime'].method != 'Clear') {
+                        await AdminProductBulkEdit.restockTimeInput.fill(changes['restockTime'].value);
+                    } 
+                }
+
+                if (changes['tags'] != null) {
+                    await AdminProductBulkEdit.changeTagsCheckbox.click();
+                    await AdminProductBulkEdit.tagsChangeMethodDropdown.click();
+                    await (await AdminProductBulkEdit.getDropdownEntry(changes['tags'].method)).click();
+                    if (changes['tags'].method != 'Clear') {
+                        await AdminProductBulkEdit.tagsInput.click();
+                        const responsePromise = AdminProductBulkEdit.page.waitForResponse(response =>
+                            response.url().includes(`search/tag`) && response.status() === 200 && response.request().method() === 'POST'
+                        );
+                        await AdminProductBulkEdit.tagsInput.fill(changes['tags'].value);
+
+                        // Wait for the search call to be completed
+                        const response = await responsePromise;
+                        expect(response.ok()).toBeTruthy();
+                        const searchResult = await AdminProductBulkEdit.getDropdownEntry(changes['tags'].value);
+                        await searchResult.click();
+                    } 
+                }
+
+                if (changes['saleschannel'] != null) {
+                    await AdminProductBulkEdit.changeSalesChannelCheckbox.click();
+                    await AdminProductBulkEdit.salesChannelChangeMethodDropdown.click();
+                    await (await AdminProductBulkEdit.getDropdownEntry(changes['saleschannel'].method)).click();
+                    if (changes['saleschannel'].method != 'Clear') {
+                        await AdminProductBulkEdit.salesChannelInput.click();
+                        const responsePromise = AdminProductBulkEdit.page.waitForResponse(response =>
+                            response.url().includes(`search/sales-channel`) && response.status() === 200 && response.request().method() === 'POST'
+                        );
+                        await AdminProductBulkEdit.salesChannelInput.fill(changes['saleschannel'].value);
+
+                        // Wait for the search call to be completed
+                        const response = await responsePromise;
+                        expect(response.ok()).toBeTruthy();
+                        const searchResult = await AdminProductBulkEdit.getDropdownEntry(changes['saleschannel'].value);
+                        await searchResult.click();
```

Comment:
I don't know if that leads to flakiness because you are dependent on the search. And maybe the test will be slow. I'm not sure.

Another solution could be:

// AdminCustomFieldCreate Page Object

```
this.assignToSelectionList = page.locator('.sw-field')
            .filter({ hasText: 'Assign To' })
            .locator('.sw-select__selection');
        this.resultAssignToPopoverItemList = page.locator('.sw-select-result-list__content').getByRole('listitem');
```

// Platform Test - CreateCustomFieldSet
```
await AdminCustomFieldCreate.assignToSelectionList.click();
await AdminCustomFieldCreate.resultAssignToPopoverItemList.getByText('Products').click();
```

Reply:
Doesn't this only pick an entry from the dropdown list? What if the entry, we want, isn't on that list? It is limited to show only a part of all entries

Reply:
`await AdminCustomFieldCreate.resultAssignToPopoverItemList.getByText('Products').click();`

with that you can decide in your test, what entity you want to pick. In my example I wanted the "Products" entity.